resilience4j2/src/main/java/com/linecorp/armeria/resilience4j/circuitbreaker/client/Resilience4JCircuitBreakerClientHandler.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.resilience4j.circuitbreaker.client;
+
+import static java.util.Objects.requireNonNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.client.UnprocessedRequestException;
+import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerClientHandler;
+import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerMapping;
+import com.linecorp.armeria.client.circuitbreaker.ClientCircuitBreakerGenerator;
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.circuitbreaker.CircuitBreakerCallback;
+
+import io.github.resilience4j.circuitbreaker.CallNotPermittedException;
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+
+/**
+ * A {@link CircuitBreakerClientHandler} implementation for use with Resilience4j's {@link CircuitBreaker}.
+ *
+ * <pre>{@code
+ * // for HttpRequest
+ * CircuitBreakerRule rule = CircuitBreakerRule.onStatusClass(HttpStatusClass.SERVER_ERROR);
+ * CircuitBreakerRegistry registry = CircuitBreakerRegistry.ofDefaults();
+ * Resilience4jCircuitBreakerMapping mapping =
+ *     Resilience4jCircuitBreakerMapping.builder()
+ *                                      .registry(registry)
+ *                                      .perHost()
+ *                                      .build();
+ * WebClient.builder()
+ *          .decorator(CircuitBreakerClient.newDecorator(
+ *              Resilience4JCircuitBreakerClientHandler.of(mapping), rule))
+ *          ...
+ *
+ * // for RpcRequest
+ * CircuitBreakerRuleWithContent rule = ...;
+ * CircuitBreakerRegistry registry = CircuitBreakerRegistry.ofDefaults();
+ * Resilience4jCircuitBreakerMapping mapping =
+ *     Resilience4jCircuitBreakerMapping.builder()
+ *                                      .registry(registry)
+ *                                      .perHost()
+ *                                      .build();
+ * ThriftClients.builder("localhost")
```

Comment:
nit: A scheme is necessary. 
```suggestion
 * ThriftClients.builder("http://thrift.api.com")
```